force-app/main/default/classes/RD2_ChangeHistoryService.cls

```diff
@@ -52,6 +52,49 @@ public inherited sharing class RD2_ChangeHistoryService {
         set;
     }
 
+    @TestVisible
+    private RD2_ChangeHistorySelector changeHistorySelector {
+        get {
+            if (changeHistorySelector == null) {
+                changeHistorySelector = new RD2_ChangeHistorySelector();
+            }
+            return changeHistorySelector;
+        }
+        set;
+    }
+
+    public RD2_ChangeHistoryView getChangeHistoryView(Id recurringDonationId, Integer recordLimit, String filter) {
+        if(!UTIL_Permissions.canRead(String.valueOf(RecurringDonationChangeHistory__c.SObjectType))) {
+            throw new UTIL_Permissions.InsufficientPermissionException(Label.commonPermissionErrorMessage);
+        }
+
+        Integer limitPlusOne = recordLimit + 1;
+        changeHistorySelector.withRecordLimit(limitPlusOne);
+
+        if (filter != null) {
+            changeHistorySelector.withChangeTypeFilter(filter);
+        }
+
+        List<RecurringDonationChangeHistory__c> changes = changeHistorySelector.getChangeHistories(recurringDonationId);
+        RD2_ChangeHistoryView changeHistoryView = buildView(changes, recordLimit);
+
+        return changeHistoryView;
+    }
+
+    private RD2_ChangeHistoryView buildView(List<RecurringDonationChangeHistory__c> changes, Integer recordLimit) {
+        RD2_ChangeHistoryView changeHistoryView = new RD2_ChangeHistoryView();
+
+        changeHistoryView.settingEnabled = this.isChangeHistoryEnabled;
```

Comment:
I think we should also check if RD2 is enabled or not to create a universal experience. Currently the other 2 components has the validation check for it, but for Change History, we only check if the Change History is enabled or not.

![Screen Shot 2021-08-10 at 3 14 13 PM](https://user-images.githubusercontent.com/57271120/128941883-6279ab2d-3c5a-4432-9b11-b43dd398bb4a.png)